### DISCUSSION TITLE
Alpha theme customization

### DISCUSSION THREAD

**Discussion by genebean** [nix-community/nixvim#1658] [Q&A]
I am working to port my neovim + lazy config to nixvim and can't quite figure out how I am supposed to translate this code:

```lua
return {
  "goolord/alpha-nvim",
  dependencies = {
    "nvim-tree/nvim-web-devicons",
  },

  config = function()
    local alpha = require("alpha")
    local dashboard = require("alpha.themes.startify")

    dashboard.section.header.val = {
      [[                                                                       ]],
      [[                                                                       ]],
      [[                                                                       ]],
      [[                                                                       ]],
      [[                                                                     ]],
      [[       ████ ██████           █████      ██                     ]],
      [[      ███████████             █████                             ]],
      [[      █████████ ███████████████████ ███   ███████████   ]],
      [[     █████████  ███    █████████████ █████ ██████████████   ]],
      [[    █████████ ██████████ █████████ █████ █████ ████ █████   ]],
      [[  ███████████ ███    ███ █████████ █████ █████ ████ █████  ]],
      [[ ██████  █████████████████████ ████ █████ █████ ████ ██████ ]],
      [[                                                                       ]],
      [[                                                                       ]],
      [[                                                                       ]],
    }

    alpha.setup(dashboard.opts)

    vim.cmd('autocmd User AlphaReady Neotree show')
  end,
}
```

I am hoping someone here can help me understand how this should be converted.

**Comment by MattSturgeon** [comment]
That config is basically loading a default theme, modifying it slightly and then passing it to `alpha`. It also triggers an autocmd.

Our config options aren't really optimized for this kinda imperative setup. Currently they can _either_ assign a default theme with [plugins.alpha.theme](https://nix-community.github.io/nixvim/plugins/alpha/index.html#pluginsalphatheme) _or_ define a custom theme using [plugins.alpha.layout](https://nix-community.github.io/nixvim/plugins/alpha/layout.html#pluginsalphalayout).

The situation might improve when we refactor the plugin to use a rfc42-style `settings` option, but for now I'd recommend avoiding the `plugins.alpha` options and instead manually adding your above config to `extraConfigLua` and adding `pkgs.vimPlugins.alpha-nvim` to `extraPlugins`.

Looks like @GaetanLepage was the last to refactor the plugin, so he might have a better suggestion as he'll be more familiar with it.

**Reply by genebean** [reply]
Thanks, that gets me unstuck. FWIW, this kind of setup is straight out of the docs for alpha at https://github.com/goolord/alpha-nvim/blob/main/doc/alpha.txt
